{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 213,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "import scipy\n",
    "import sklearn\n",
    "import os\n",
    "import subprocess\n",
    "import json\n",
    "import sys\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [],
   "source": [
    "#file names, should try to automate more\n",
    "VCF_FILE_NAME = 'ERR6634978_q20.recode.vcf.gz'\n",
    "VCF_PATH = '../../../data/processed/freebayesVCF'\n",
    "VCF_FILE_PATH = os.path.join(VCF_PATH, VCF_FILE_NAME)\n",
    "JSON_FILE_PATH = os.path.join(VCF_PATH, 'results/ERR6634978.results.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [],
   "source": [
    "#tried to recognise whether if the vcf file is in gz format, if not, compress it\n",
    "# import re\n",
    "\n",
    "# if re.match('*vcf$',VCF_FILE_NAME):\n",
    "#     print('Please input .vcf.gz format file compresed using bgzip')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#running tb-profiler to identify drug resistance for the sample\n",
    "#need to be run in tb-profiler environment, somehow failed to download tb-profiler in ml-s7, so far this part is run out side python in bash\n",
    "subprocess.call(['tb-profiler', 'vcf_profile', VCF_FILE_PATH])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#read in the drug resistance labeled file generated from tb-profiler\n",
    "with open(JSON_FILE_PATH, 'r') as f:\n",
    "    json_labelling = json.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Could not find conda environment: tb-profiler\n",
      "You can list all discoverable environments with `conda info --envs`.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#get the reference read and alternative read count, changing environment is somehow not working in python\n",
    "env_command = 'eval \"$(conda shell.bash hook)\"; conda activate tb-profiler'\n",
    "subprocess.run(env_command, shell=True)\n",
    "\n",
    "with open('ROAO_proportion.csv', 'w') as f:\n",
    "    subprocess.run(['bcftools', 'query', '-f', '[%POS,%RO,%AO\\n]', VCF_FILE_PATH], shell=False, stdout=f, text=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4\n",
      "3\n"
     ]
    }
   ],
   "source": [
    "#count how many column there is in the ROAO_proportion.csv file this is needed in order to read the csv in a a panda dataframe\n",
    "import csv\n",
    "\n",
    "f = open('ROAO_proportion.csv')\n",
    "\n",
    "data = []\n",
    "num_col=0\n",
    "for line in f:\n",
    "    data_line = line.rstrip().split('\\n')\n",
    "    data_line = data_line[0].split(',')\n",
    "    data_line = [int(i) for i in data_line]\n",
    "    if len(data_line) > num_col:\n",
    "        num_col = len(data_line)\n",
    "    data.append(data_line)\n",
    "\n",
    "print(num_col)\n",
    "num_RO = num_col - 1\n",
    "print(num_RO)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for reads in json_labelling['dr_variants']:\n",
    "    print(reads['genome_pos'])\n",
    "    for i in reads['drugs']:\n",
    "        print(i['drug'])\n",
    "\n",
    "for reads in json_labelling['other_variants']:\n",
    "    print(reads['genome_pos'])\n",
    "    print(reads['gene_associated_drugs'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [],
   "source": [
    "#reading csv as pandas dataframe and create a numpy array version of it\n",
    "df = pd.read_csv('ROAO_proportion.csv', names=range(num_col)) #doesn't include the 4th colum\n",
    "df.fillna(value=0, inplace=True)\n",
    "df[num_col] = None #create 4th column here as a empty column of zero, this column is for the drug resistance info\n",
    "df_np = df.to_numpy()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hit:  761155\n",
      "hit:  2155168\n",
      "hit:  4247429\n",
      "hit:  7362\n",
      "hit:  7585\n",
      "hit:  9304\n",
      "hit:  764817\n",
      "hit:  764995\n",
      "hit:  781395\n",
      "hit:  1416222\n",
      "hit:  1471659\n",
      "hit:  1917972\n",
      "hit:  3073868\n",
      "hit:  3086788\n",
      "hit:  4239763\n",
      "hit:  4242643\n",
      "hit:  4408156\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#numpy array and json file compared against each other in terms of genome position and \n",
    "#the drug resistance info is added on to the 4th column of the numpy array if there is a match in genome position\n",
    "for ind, pos in enumerate(df_np[:,0]): \n",
    "    for reads in json_labelling['dr_variants']:\n",
    "        if pos == reads['genome_pos']:\n",
    "            # print('hit: ', pos)\n",
    "            for i in reads['drugs']:\n",
    "                df_np[ind, 4] = [i['drug']]\n",
    "\n",
    "for ind, pos in enumerate(df_np[:,0]):\n",
    "    for reads in json_labelling['other_variants']:\n",
    "        if pos == reads['genome_pos']:\n",
    "            # print('hit: ', pos)\n",
    "            df_np[ind, 4] = reads['gene_associated_drugs']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [],
   "source": [
    "#training gaussianMixture Model\n",
    "\n",
    "from sklearn.mixture import GaussianMixture\n",
    "\n",
    "model_input = df_np[:,1:num_col]\n",
    "gm = GaussianMixture(n_components=num_RO, random_state=0).fit(model_input)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'strain1': [], 'strain2': [], 'strain3': []}"
      ]
     },
     "execution_count": 245,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#create number of strain number of object in a dictionary\n",
    "resistance_dict = {}\n",
    "\n",
    "for x in range(1,num_RO+1):\n",
    "    resistance_dict['strain'+ str(x)] = []\n",
    "\n",
    "resistance_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['fluoroquinolones', 'levofloxacin', 'moxifloxacin', 'ciprofloxacin', 'ofloxacin'] added\n",
      "['fluoroquinolones', 'levofloxacin', 'moxifloxacin', 'ciprofloxacin', 'ofloxacin'] added\n",
      "['fluoroquinolones', 'levofloxacin', 'moxifloxacin', 'ciprofloxacin', 'ofloxacin'] added\n",
      "['rifampicin'] added\n",
      "['rifampicin'] added\n",
      "['rifampicin'] added\n",
      "['streptomycin'] added\n",
      "['ethambutol'] added\n",
      "['amikacin', 'aminoglycosides', 'kanamycin', 'streptomycin', 'capreomycin'] added\n",
      "['capreomycin'] added\n",
      "['isoniazid'] added\n",
      "['para-aminosalicylic_acid'] added\n",
      "['cycloserine'] added\n",
      "['ethambutol'] added\n",
      "['ethambutol'] added\n",
      "['ethambutol'] added\n",
      "['streptomycin'] added\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/mnt/storage7/lwang/miniconda3/envs/ml-s7/lib/python3.10/site-packages/sklearn/base.py:561: FutureWarning: Arrays of bytes/strings is being converted to decimal numbers if dtype='numeric'. This behavior is deprecated in 0.24 and will be removed in 1.1 (renaming of 0.26). Please convert your data to numeric values explicitly instead.\n",
      "  X = check_array(X, **check_params)\n",
      "/mnt/storage7/lwang/miniconda3/envs/ml-s7/lib/python3.10/site-packages/sklearn/base.py:561: FutureWarning: Arrays of bytes/strings is being converted to decimal numbers if dtype='numeric'. This behavior is deprecated in 0.24 and will be removed in 1.1 (renaming of 0.26). Please convert your data to numeric values explicitly instead.\n",
      "  X = check_array(X, **check_params)\n",
      "/mnt/storage7/lwang/miniconda3/envs/ml-s7/lib/python3.10/site-packages/sklearn/base.py:561: FutureWarning: Arrays of bytes/strings is being converted to decimal numbers if dtype='numeric'. This behavior is deprecated in 0.24 and will be removed in 1.1 (renaming of 0.26). Please convert your data to numeric values explicitly instead.\n",
      "  X = check_array(X, **check_params)\n",
      "/mnt/storage7/lwang/miniconda3/envs/ml-s7/lib/python3.10/site-packages/sklearn/base.py:561: FutureWarning: Arrays of bytes/strings is being converted to decimal numbers if dtype='numeric'. This behavior is deprecated in 0.24 and will be removed in 1.1 (renaming of 0.26). Please convert your data to numeric values explicitly instead.\n",
      "  X = check_array(X, **check_params)\n",
      "/mnt/storage7/lwang/miniconda3/envs/ml-s7/lib/python3.10/site-packages/sklearn/base.py:561: FutureWarning: Arrays of bytes/strings is being converted to decimal numbers if dtype='numeric'. This behavior is deprecated in 0.24 and will be removed in 1.1 (renaming of 0.26). Please convert your data to numeric values explicitly instead.\n",
      "  X = check_array(X, **check_params)\n",
      "/mnt/storage7/lwang/miniconda3/envs/ml-s7/lib/python3.10/site-packages/sklearn/base.py:561: FutureWarning: Arrays of bytes/strings is being converted to decimal numbers if dtype='numeric'. This behavior is deprecated in 0.24 and will be removed in 1.1 (renaming of 0.26). Please convert your data to numeric values explicitly instead.\n",
      "  X = check_array(X, **check_params)\n",
      "/mnt/storage7/lwang/miniconda3/envs/ml-s7/lib/python3.10/site-packages/sklearn/base.py:561: FutureWarning: Arrays of bytes/strings is being converted to decimal numbers if dtype='numeric'. This behavior is deprecated in 0.24 and will be removed in 1.1 (renaming of 0.26). Please convert your data to numeric values explicitly instead.\n",
      "  X = check_array(X, **check_params)\n",
      "/mnt/storage7/lwang/miniconda3/envs/ml-s7/lib/python3.10/site-packages/sklearn/base.py:561: FutureWarning: Arrays of bytes/strings is being converted to decimal numbers if dtype='numeric'. This behavior is deprecated in 0.24 and will be removed in 1.1 (renaming of 0.26). Please convert your data to numeric values explicitly instead.\n",
      "  X = check_array(X, **check_params)\n",
      "/mnt/storage7/lwang/miniconda3/envs/ml-s7/lib/python3.10/site-packages/sklearn/base.py:561: FutureWarning: Arrays of bytes/strings is being converted to decimal numbers if dtype='numeric'. This behavior is deprecated in 0.24 and will be removed in 1.1 (renaming of 0.26). Please convert your data to numeric values explicitly instead.\n",
      "  X = check_array(X, **check_params)\n",
      "/mnt/storage7/lwang/miniconda3/envs/ml-s7/lib/python3.10/site-packages/sklearn/base.py:561: FutureWarning: Arrays of bytes/strings is being converted to decimal numbers if dtype='numeric'. This behavior is deprecated in 0.24 and will be removed in 1.1 (renaming of 0.26). Please convert your data to numeric values explicitly instead.\n",
      "  X = check_array(X, **check_params)\n",
      "/mnt/storage7/lwang/miniconda3/envs/ml-s7/lib/python3.10/site-packages/sklearn/base.py:561: FutureWarning: Arrays of bytes/strings is being converted to decimal numbers if dtype='numeric'. This behavior is deprecated in 0.24 and will be removed in 1.1 (renaming of 0.26). Please convert your data to numeric values explicitly instead.\n",
      "  X = check_array(X, **check_params)\n",
      "/mnt/storage7/lwang/miniconda3/envs/ml-s7/lib/python3.10/site-packages/sklearn/base.py:561: FutureWarning: Arrays of bytes/strings is being converted to decimal numbers if dtype='numeric'. This behavior is deprecated in 0.24 and will be removed in 1.1 (renaming of 0.26). Please convert your data to numeric values explicitly instead.\n",
      "  X = check_array(X, **check_params)\n",
      "/mnt/storage7/lwang/miniconda3/envs/ml-s7/lib/python3.10/site-packages/sklearn/base.py:561: FutureWarning: Arrays of bytes/strings is being converted to decimal numbers if dtype='numeric'. This behavior is deprecated in 0.24 and will be removed in 1.1 (renaming of 0.26). Please convert your data to numeric values explicitly instead.\n",
      "  X = check_array(X, **check_params)\n",
      "/mnt/storage7/lwang/miniconda3/envs/ml-s7/lib/python3.10/site-packages/sklearn/base.py:561: FutureWarning: Arrays of bytes/strings is being converted to decimal numbers if dtype='numeric'. This behavior is deprecated in 0.24 and will be removed in 1.1 (renaming of 0.26). Please convert your data to numeric values explicitly instead.\n",
      "  X = check_array(X, **check_params)\n",
      "/mnt/storage7/lwang/miniconda3/envs/ml-s7/lib/python3.10/site-packages/sklearn/base.py:561: FutureWarning: Arrays of bytes/strings is being converted to decimal numbers if dtype='numeric'. This behavior is deprecated in 0.24 and will be removed in 1.1 (renaming of 0.26). Please convert your data to numeric values explicitly instead.\n",
      "  X = check_array(X, **check_params)\n",
      "/mnt/storage7/lwang/miniconda3/envs/ml-s7/lib/python3.10/site-packages/sklearn/base.py:561: FutureWarning: Arrays of bytes/strings is being converted to decimal numbers if dtype='numeric'. This behavior is deprecated in 0.24 and will be removed in 1.1 (renaming of 0.26). Please convert your data to numeric values explicitly instead.\n",
      "  X = check_array(X, **check_params)\n",
      "/mnt/storage7/lwang/miniconda3/envs/ml-s7/lib/python3.10/site-packages/sklearn/base.py:561: FutureWarning: Arrays of bytes/strings is being converted to decimal numbers if dtype='numeric'. This behavior is deprecated in 0.24 and will be removed in 1.1 (renaming of 0.26). Please convert your data to numeric values explicitly instead.\n",
      "  X = check_array(X, **check_params)\n"
     ]
    }
   ],
   "source": [
    "#reads are assigned to different strains according to their ROAO by the gaussianMixture model\n",
    "#and added to the corresponding strain item in dictionary\n",
    "\n",
    "for x in df_np:\n",
    "    if x[num_col] != None:\n",
    "        output = gm.predict([x[1:num_col]])\n",
    "        # print(output)\n",
    "\n",
    "        for i in range(1,num_RO+1):\n",
    "            if output == [i-1]:\n",
    "                # if x[num_col] in resistance_dict['strain'+ str(i)]:\n",
    "                #     continue\n",
    "                resistance_dict['strain'+ str(i)].extend(x[num_col])\n",
    "                # print(x[num_col], 'added')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 249,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>strain1</th>\n",
       "      <th>strain2</th>\n",
       "      <th>strain3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ethambutol</td>\n",
       "      <td>amikacin</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>aminoglycosides</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>capreomycin</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ciprofloxacin</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cycloserine</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ethambutol</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NaN</td>\n",
       "      <td>fluoroquinolones</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NaN</td>\n",
       "      <td>isoniazid</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NaN</td>\n",
       "      <td>kanamycin</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NaN</td>\n",
       "      <td>levofloxacin</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NaN</td>\n",
       "      <td>moxifloxacin</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ofloxacin</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NaN</td>\n",
       "      <td>para-aminosalicylic_acid</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NaN</td>\n",
       "      <td>rifampicin</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>NaN</td>\n",
       "      <td>streptomycin</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       strain1                   strain2  strain3\n",
       "0   ethambutol                  amikacin      NaN\n",
       "1          NaN           aminoglycosides      NaN\n",
       "2          NaN               capreomycin      NaN\n",
       "3          NaN             ciprofloxacin      NaN\n",
       "4          NaN               cycloserine      NaN\n",
       "5          NaN                ethambutol      NaN\n",
       "6          NaN          fluoroquinolones      NaN\n",
       "7          NaN                 isoniazid      NaN\n",
       "8          NaN                 kanamycin      NaN\n",
       "9          NaN              levofloxacin      NaN\n",
       "10         NaN              moxifloxacin      NaN\n",
       "11         NaN                 ofloxacin      NaN\n",
       "12         NaN  para-aminosalicylic_acid      NaN\n",
       "13         NaN                rifampicin      NaN\n",
       "14         NaN              streptomycin      NaN"
      ]
     },
     "execution_count": 249,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#remove ducplicate items and show the dictionary as a df\n",
    "for x in range(1,num_RO+1):\n",
    "    resistance_dict['strain'+ str(x)] = np.unique(resistance_dict['strain'+ str(x)])\n",
    "\n",
    "pd.DataFrame(dict([ (k,pd.Series(v)) for k,v in resistance_dict.items() ]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<BarContainer object of 1171 artists>"
      ]
     },
     "execution_count": 262,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXkAAAD4CAYAAAAJmJb0AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/YYfK9AAAACXBIWXMAAAsTAAALEwEAmpwYAAAQKElEQVR4nO3db6xkd13H8ffHXYv8tUAvBHZbd01qdR/wp6yl+Betwm4hrCY8aPlTQJpNIzWoD+wSEhPDExA1hFDYrLUiijQGKqywUAz+IQYK3SqULmVh2WJ7abVbUSSQWBa+PpizMEzn3nvm3rn3zvz2/Upu7pzf+c2Z73dm9nPPnDMzm6pCktSmH9rsAiRJ68eQl6SGGfKS1DBDXpIaZshLUsO2btYNn3feebVjx47NunlJmku33377g1W10Hf+poX8jh07OHr06GbdvCTNpST/Psl8D9dIUsMMeUlqmCEvSQ0z5CWpYYa8JDXMkJekhq0Y8kluTPJAkjuXWJ8kb01yIskdSS6efpmSpNXosyf/TmDPMuv3Ahd2P/uBd6y9LEnSNKwY8lX1ceBry0zZB7yrBm4Fzk3ylGkVKElavWl84nUbcO/Q8mI3dv/oxCT7Geztc8EFF6z6Bncc+ND3Ln/ljS9Ycnx4edjwujOXh7fTt4Yz1xndVt/bGVff6Lzh649bXqqu5W5reFvjah7d3rjbH6153HWG+1mqv6UsV+9yvS7X0+i6vvPG9brS86fPc3S0hnF9jqt13PYmeV4tdZ3ROX2eP0vdzrjxPtdZ7v5c6b5e6rkwqu/9vdT2+/S60m1MmjdrMY0TrxkzNva/m6qqQ1W1u6p2Lyz0/uqFpiz1h2etc5e73krL61nLRjpT41rut7X2uZbrb8Zzo8+6jXzs+zyGs1DnPJlGyC8C5w8tbwfum8J2N9RaA2Kl60663Y16wm72P4zNvn3Nr/V87kzjD/40t7MW0wj5w8BV3btsLgW+XlUPO1Sznjb7TpSmxb3Upa33q6RW7+M+b6F8D/BJ4KIki0leneSaJNd0U44AJ4ETwJ8Cv7lu1WqmtfqPRJpnK554raorV1hfwGumVpFmwkafHGqJf+w2n4/B9531n3j1yTBd3p/SbDnrQ16SWmbIS1LDDHlJ2gCbdSjTkJekhhnyUgM84a2lGPKS1DBDXpIaZshLUsMMeUlqmCEvSQ1rJuR9d4EkPVwzIS9JejhDXpIaZshLUsMMeUlqmCEvSQ0z5CWpYYa8JDXMkJekhhnyktQwQ16SGmbIS1LDDHlJapghL03IL8PTPDHkJalhhrwkNcyQl6SGGfKS1DBDfk558k9SH4a8JDXMkJekhvUK+SR7khxPciLJgTHrfzTJ3yX5bJJjSV41/VIlSZNaMeSTbAGuB/YCu4Ark+wamfYa4PNV9XTgucAfJzlnyrVKkibUZ0/+EuBEVZ2sqoeAm4B9I3MKeGySAI8BvgacnmqlkqSJ9Qn5bcC9Q8uL3diwtwE/BdwHfA54bVV9d3RDSfYnOZrk6KlTp1ZZsiSprz4hnzFjNbL8fOAzwFOBZwBvS/K4h12p6lBV7a6q3QsLCxOWKkmaVJ+QXwTOH1rezmCPfdirgJtr4ARwN/CT0ylRkrRafUL+NuDCJDu7k6lXAIdH5twDXAaQ5MnARcDJaRYqaeP5obv5t3WlCVV1Osm1wC3AFuDGqjqW5Jpu/UHgDcA7k3yOweGd66rqwXWsW5LUw4ohD1BVR4AjI2MHhy7fBzxvuqVJktbKT7xKUsMMeUlqmCEvSQ0z5CWpYYa8JDXMkJekhhnyktQwQ16SGmbIS1LDDHlJapghL0kNM+QlqWGGvCQ1zJCXpIYZ8pLUMENekhpmyEtSwwx5SWqYIS9JDTPkJalhhrwkNcyQl6SGGfKS1DBDXnNhx4EPbXYJ0lwy5CWpYYa8JDXMkNdMOHM4ps9hGQ/dTIf349nBkJekhhnyktQwQ16SGmbIS1LDDPmGeCJN0qheIZ9kT5LjSU4kObDEnOcm+UySY0n+ebplSpJWY+tKE5JsAa4HfhVYBG5LcriqPj8051zg7cCeqronyZPWqV5J0gT67MlfApyoqpNV9RBwE7BvZM5LgJur6h6AqnpgumVKklajT8hvA+4dWl7sxob9BPD4JP+U5PYkV02rQEnS6q14uAbImLEas51nAZcBjwQ+meTWqvriD2wo2Q/sB7jgggsmr1aSNJE+e/KLwPlDy9uB+8bM+UhVfbOqHgQ+Djx9dENVdaiqdlfV7oWFhdXWLEnqqU/I3wZcmGRnknOAK4DDI3M+APx8kq1JHgU8G7hruqVKkia14uGaqjqd5FrgFmALcGNVHUtyTbf+YFXdleQjwB3Ad4EbqurO9SxckrSyPsfkqaojwJGRsYMjy28G3jy90iRJa+UnXiWpYYa8JDXMkJekhhnyktQwQ17SD/DbTNtiyEtSwwx5SWqYId+DL18lzStDXpIaZshLUsMMeUlqmCEvbRDP7WgzGPKS1DBDXpIaZshLUsMMeUlqmCEvSQ0z5CWpYYa8pLnl21JXZshPyCfVbPBxkPox5CWpYYa8JDXMkJekhhnyktQwQ15nFU/Y6mxjyEtSwwx5SWqYIb8GvvSXNOsMeUlqmCG/Cu7BS5oXhrwkNcyQ16bzlZG0fgz5CRhGm8P7XVq9XiGfZE+S40lOJDmwzLyfTvKdJC+eXomSpNVaMeSTbAGuB/YCu4Ark+xaYt6bgFumXaQkaXX67MlfApyoqpNV9RBwE7BvzLzfAt4HPDDF+iRpTc72w319Qn4bcO/Q8mI39j1JtgG/DhxcbkNJ9ic5muToqVOnJq1Vyzjbn8iSxusT8hkzViPLbwGuq6rvLLehqjpUVburavfCwkLPEjeegSmpFVt7zFkEzh9a3g7cNzJnN3BTEoDzgMuTnK6q90+jSEnS6vQJ+duAC5PsBL4KXAG8ZHhCVe08cznJO4EPGvCStPlWDPmqOp3kWgbvmtkC3FhVx5Jc061f9ji8JGnz9NmTp6qOAEdGxsaGe1W9cu1l6WzkuRBp+vzEqzaUQa5J+ZxZG0Ne2gAGlTaLIS9JDTPkl+Cel6QWGPKSmuXOmiEvTYVholllyM85w0XScgz5OWbAS1qJIS/prHU27CgZ8pLUMENekhpmyC/jbHgpJ6lthrwkNcyQl6SGGfKS1DBDXpIaZshLUsMMeUlqmCEvSQ0z5DXX/CyDtDxDXpIaZshLUsMM+RV4OEDSPDPkJalhhrwkNcyQl6SGGfKS1DBDXpIaZshLUsMMeUlnjbPxLdGGvCQ1zJDXks7GvZ5JeR9p1vUK+SR7khxPciLJgTHrX5rkju7nE0mePv1SJUmTWjHkk2wBrgf2AruAK5PsGpl2N/CLVfU04A3AoWkXKkmaXJ89+UuAE1V1sqoeAm4C9g1PqKpPVNV/d4u3AtunW6YkaTX6hPw24N6h5cVubCmvBj48bkWS/UmOJjl66tSp/lVKklalT8hnzFiNnZj8EoOQv27c+qo6VFW7q2r3wsJC/yolSauytcecReD8oeXtwH2jk5I8DbgB2FtV/zWd8qTp8F0wOlv12ZO/Dbgwyc4k5wBXAIeHJyS5ALgZeHlVfXH6ZUrSdE36h39edxRW3JOvqtNJrgVuAbYAN1bVsSTXdOsPAr8PPBF4exKA01W1e/3KljSvoaON1edwDVV1BDgyMnZw6PLVwNXTLU2StFZ+4lWSGmbIS1LDDHlJapghL0kNM+QlqWGGvCQ1zJCXpIYZ8pLUMENekhpmyEtSwwx5SWqYIS9JDTPkJalhhrwkNcyQl6SGGfKS1DBDXpIaZshLUsMMeUlqmCEvSQ0z5CWpYYa8JDXMkJekhhnyktQwQ16SGmbIS1LDDHlJapghL0kNM+QlqWGGvCQ1zJCXpIYZ8pLUMENekhrWK+ST7ElyPMmJJAfGrE+St3br70hy8fRLlSRNasWQT7IFuB7YC+wCrkyya2TaXuDC7mc/8I4p1ylJWoU+e/KXACeq6mRVPQTcBOwbmbMPeFcN3Aqcm+QpU65VkjShVNXyE5IXA3uq6upu+eXAs6vq2qE5HwTeWFX/0i1/DLiuqo6ObGs/gz19gIuA42uo/TzgwTVcf9bYz+xqqRewn1m3Uj8/VlULfTe2tcecjBkb/cvQZw5VdQg41OM2Vy4qOVpVu6exrVlgP7OrpV7AfmbdtPvpc7hmETh/aHk7cN8q5kiSNlifkL8NuDDJziTnAFcAh0fmHAau6t5lcynw9aq6f8q1SpImtOLhmqo6neRa4BZgC3BjVR1Lck23/iBwBLgcOAF8C3jV+pX8PVM57DND7Gd2tdQL2M+sm2o/K554lSTNLz/xKkkNM+QlqWFzF/IrfcXCLEpyfpJ/THJXkmNJXtuNPyHJ3yf5Uvf78UPXeV3X4/Ekz9+86sdLsiXJv3WfkZjrXgCSnJvkvUm+0D1Oz5nnnpL8TvdcuzPJe5L8yDz1k+TGJA8kuXNobOL6kzwryee6dW9NMu7t3utuiX7e3D3f7kjyt0nOHVo3vX6qam5+GJz4/TLw48A5wGeBXZtdV4+6nwJc3F1+LPBFBl8R8YfAgW78APCm7vKurrdHADu7nrdsdh8jPf0u8NfAB7vlue2lq/MvgKu7y+cA585rT8A24G7gkd3y3wCvnKd+gF8ALgbuHBqbuH7g08BzGHyW58PA3hnq53nA1u7ym9arn3nbk+/zFQszp6rur6p/7S5/A7iLwT/EfQzChe73r3WX9wE3VdX/VdXdDN61dMmGFr2MJNuBFwA3DA3PZS8ASR7H4B/hnwFU1UNV9T/McU8M3jn3yCRbgUcx+NzK3PRTVR8HvjYyPFH93VerPK6qPlmDhHzX0HU21Lh+quqjVXW6W7yVweeLYMr9zFvIbwPuHVpe7MbmRpIdwDOBTwFPru7zBN3vJ3XTZr3PtwC/B3x3aGxee4HBK8NTwJ93h6BuSPJo5rSnqvoq8EfAPcD9DD638lHmtJ8hk9a/rbs8Oj6LfoPBnjlMuZ95C/leX58wq5I8Bngf8NtV9b/LTR0zNhN9Jnkh8EBV3d73KmPGZqKXIVsZvJR+R1U9E/gmg8MBS5npnrpj1fsYvNR/KvDoJC9b7ipjxmamnx6Wqn8u+kryeuA08O4zQ2OmrbqfeQv5uf36hCQ/zCDg311VN3fD/9m9BKP7/UA3Pst9/izwoiRfYXC47JeT/BXz2csZi8BiVX2qW34vg9Cf155+Bbi7qk5V1beBm4GfYX77OWPS+hf5/iGQ4fGZkeQVwAuBl3aHYGDK/cxbyPf5ioWZ050B/zPgrqr6k6FVh4FXdJdfAXxgaPyKJI9IspPB9/R/eqPqXU5Vva6qtlfVDgb3/z9U1cuYw17OqKr/AO5NclE3dBnweea3p3uAS5M8qnvuXcbgPNC89nPGRPV3h3S+keTS7n64aug6my7JHuA64EVV9a2hVdPtZzPONK/xLPXlDN6d8mXg9ZtdT8+af47By6o7gM90P5cDTwQ+Bnyp+/2Eoeu8vuvxOJv0joAefT2X77+7Zt57eQZwtHuM3g88fp57Av4A+AJwJ/CXDN6pMTf9AO9hcD7h2wz2YF+9mvqB3d198GXgbXSf8p+Rfk4wOPZ+JhMOrkc/fq2BJDVs3g7XSJImYMhLUsMMeUlqmCEvSQ0z5CWpYYa8JDXMkJekhv0/Bgkx9Bw+oBUAAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "#frequency graph assuming that there is only two strains\n",
    "df_2component = pd.read_csv('ROAO_proportion.csv', names=range(3))\n",
    "df_2component['proportion'] = df_2component[2]/(df_2component[1]+df_2component[2]) # alt/(ref+alt)\n",
    "\n",
    "fig, ax= plt.subplots()\n",
    "ax.bar(np.arange(len(df_2component[2])),df_2component['proportion'])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "d6768d4663a47fff933bae7805b3d263e035a1f0db74783c60faa02117d87629"
  },
  "kernelspec": {
   "display_name": "Python 3.7.10 64-bit ('assembly': conda)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.0"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
