{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e4ad1150-bf53-4216-9142-f38693f6bbf5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "# import pathogenprofiler as pp\n",
    "from sklearn.mixture import GaussianMixture\n",
    "from sklearn.metrics import mean_squared_error, confusion_matrix, f1_score\n",
    "# import fastq2matrix as fm\n",
    "import plotly.graph_objects as go\n",
    "from plotly.subplots import make_subplots\n",
    "import json\n",
    "from scipy.stats import norm\n",
    "import subprocess\n",
    "from scipy.stats import kurtosis, skew\n",
    "import re\n",
    "import os\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a85b3407-8679-4f8b-a39b-f2c38cdfe752",
   "metadata": {},
   "outputs": [],
   "source": [
    "vcf_file = 'test_data/ERR6634978-ERR6635032-2080.vcf.gz'\n",
    "json_file = 'test_data/ERR6634978-ERR6635032-2080.results.json'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "98feed87-d6b0-4ac7-967b-93ce0b3ea9ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "VCF_FILE_PATH='/mnt/storage7/lwang/trial_tb_philippines/data/processed/seqtk/freebayesVCF/'\n",
    "#/mnt/storage7/lwang/trial_tb_philippines/data/processed/seqtk/freebayesVCF/q20\n",
    "JSON_FILE_PATH='/mnt/storage7/lwang/trial_tb_philippines/data/processed/seqtk/freebayesVCF/results'\n",
    "#/mnt/storage7/lwang/trial_tb_philippines/data/processed/seqtk/freebayesVCF/q20/results\n",
    "\n",
    "NAME_FILE='/mnt/storage7/lwang/trial_tb_philippines/data/seqtk/sample_name.txt'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "817649b6-ff4f-4c48-b1a3-a48f870320f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "model_predictions = []\n",
    "\n",
    "def model_pred(vcf_file, ratio='test'):\n",
    "    with open('mix_infection.csv', 'w') as f:\n",
    "        subprocess.run(\"bcftools view -c 1 -m2 -M2 -T ^new_exclusion.bed %s | bcftools query -f '%%POS\\\\t%%REF\\\\t%%ALT[\\\\t%%GT\\\\t%%AD\\\\n]'\" % vcf_file, shell=True, stdout=f, text=True)\n",
    "\n",
    "#count how many column there is in the ROAO_proportion.csv file this is needed in order to read the csv in a a panda dataframe\n",
    "    pos = []\n",
    "    freqs = []\n",
    "    scatter = []\n",
    "    with open('mix_infection.csv', 'r') as f:\n",
    "        for l in f:\n",
    "            row = l.strip().split()\n",
    "            ads = [int(x) for x in row[4].split(\",\")]\n",
    "            afs = [x/sum(ads) for x in ads]\n",
    "\n",
    "            pos.append(int(row[0]))\n",
    "            freqs.append([afs[1]])\n",
    "            scatter.append(ads)\n",
    "            \n",
    "        fig, ax= plt.subplots(figsize=(12, 8), ncols=1, nrows=2)\n",
    "\n",
    "        ax[0].hist(freqs, bins=200)\n",
    "        ax[0].set_ylim([0,300])\n",
    "        ax[0].set_xlim([0,1])\n",
    "        ax[0].set_ylabel('frequency')\n",
    "\n",
    "#         ax[1].hist(df_np[:,2], bins=200)\n",
    "#         ax[1].set_ylim([0,300])\n",
    "#         ax[1].set_title(\"alternative\")\n",
    "#         ax[1].set_xlim([0,500])\n",
    "\n",
    "#         ax[1].set_xlabel('number of haplotype copy')\n",
    "#         ax[1].set_ylabel('count')\n",
    "\n",
    "\n",
    "        fig.tight_layout()\n",
    "\n",
    "        plt.figure()\n",
    "        \n",
    "\n",
    "print(model_pred(vcf_file))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "91d7c26d-2adc-4314-91a5-14db17db23d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "def frac_MSE(NAME_FILE=NAME_FILE, JSON_FILE_PATH=JSON_FILE_PATH, VCF_FILE_PATH=VCF_FILE_PATH):\n",
    "    with open(NAME_FILE, 'r') as f:\n",
    "        sample_list = [line.rstrip('\\n') for line in f]\n",
    "\n",
    "    sample_list_json=[]\n",
    "    for i,x in enumerate(sample_list):\n",
    "        x = list(x)\n",
    "        x.append('.results.json')\n",
    "        string = ''.join([letter for letter in x] )\n",
    "        sample_list_json.append(string)\n",
    "    \n",
    "\n",
    "    sample_list_vcf=[]\n",
    "    for i,x in enumerate(sample_list):\n",
    "        x = list(x)\n",
    "        x.append('.vcf.gz')  #change here if using filtered/non-filtered samples\n",
    "        string = ''.join([letter for letter in x] )\n",
    "        sample_list_vcf.append(string)\n",
    "\n",
    "    ratio = []\n",
    "    for i in sample_list_json:\n",
    "        temp = i.split('-')\n",
    "        ratio.append(temp[2])\n",
    "    \n",
    "    ratio1=[]\n",
    "    for i in ratio:\n",
    "        temp = i.split('.')\n",
    "        ratio1.append(temp[0])\n",
    "\n",
    "    ratio = ratio1\n",
    "\n",
    "    tb_profiler_predictions = []\n",
    "    model_predictions = []\n",
    "\n",
    "    for j, vcf, ratio in zip(sample_list_json, sample_list_vcf, ratio):\n",
    "        JSON_FILE = os.path.join(JSON_FILE_PATH, j)\n",
    "        tb_out_, cut_off = tb_profiler_pred(JSON_FILE)\n",
    "\n",
    "        VCF_FILE = os.path.join(VCF_FILE_PATH, vcf)\n",
    "#graphing for sample SNPs(using gary's database)\n",
    "        model_out_, gm = model_pred(vcf_file, tail_cutoff=cut_off, ratio=ratio)\n",
    "\n",
    "#graphing for lineage specific snps found in the sample (using gary's database)\n",
    "        print(f'runing anchor for {ratio}')\n",
    "        anchored_snp, anchored_lineage = anchor(VCF_FILE)\n",
    "\n",
    "        anchored_array = np.concatenate((anchored_snp, anchored_lineage.reshape(-1,1)), axis=1) \n",
    "\n",
    "        lin1_ = anchored_array[anchored_array[:,2]==1]\n",
    "        lin4_ = anchored_array[anchored_array[:,2]==4]\n",
    "\n",
    "        plt.scatter(lin1_[:,0], lin1_[:,1], c='pink', alpha=0.8, label='signature_lin_1')\n",
    "        plt.scatter(lin4_[:,0], lin4_[:,1], c='green', alpha=0.7, label='signature_lin_4')\n",
    "        plt.title(f'ratio: {ratio}')\n",
    "        plt.legend(title=\"lineage specific SNPs\",loc=\"upper right\")\n",
    "\n",
    "\n",
    "\n",
    "        tb_profiler_predictions.extend(tb_out_)\n",
    "        model_predictions.extend(model_out_)\n",
    "        print(f'{ratio}, done')\n",
    "    \n",
    "    print(tb_profiler_predictions)\n",
    "    print(model_predictions)\n",
    "    MSE = mean_squared_error(tb_profiler_predictions, model_predictions, squared=False)\n",
    "\n",
    "    return MSE"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
